subprojects/logging/src/main/java/org/gradle/internal/logging/sink/AnsiConsoleUtil.java

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.logging.sink;
+
+import org.fusesource.jansi.AnsiOutputStream;
+import org.fusesource.jansi.internal.Kernel32.*;
+import org.fusesource.jansi.internal.WindowsSupport;
+
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import static org.fusesource.jansi.internal.CLibrary.STDOUT_FILENO;
+import static org.fusesource.jansi.internal.CLibrary.isatty;
+import static org.fusesource.jansi.internal.Kernel32.*;
+
+/**
+ * @see <a href="https://github.com/fusesource/jansi/issues/69">fusesource/jansi#69</a>
```

Comment:
I added comments that links to the right commit as well as highlight the copied code.